internal/auth/issuer/pam/pam_issuer.go

```diff
@@ -0,0 +1,951 @@
+		return &pamapi.TokenResponse{Error: lo.ToPtr("invalid_grant")}, nil
+	}
+
+	// Get user information from NSS
+	systemUser, err := s.pamAuthenticator.LookupUser(codeData.Username)
+	if err != nil {
+		return &pamapi.TokenResponse{Error: lo.ToPtr("invalid_grant")}, nil
+	}
+
+	// Get user groups for roles
+	groups, err := s.pamAuthenticator.GetUserGroups(systemUser)
+	if err != nil {
+		return &pamapi.TokenResponse{Error: lo.ToPtr("server_error")}, nil
+	}
+
+	// Map groups to roles and extract organizations
+	roles := s.mapGroupsToRoles(groups)
+	organizations := s.extractOrganizations(groups)
+
+	// Create identity for token generation
+	tokenGenerationRequest := authn.TokenGenerationRequest{
+		Username:      codeData.Username,
+		UID:           codeData.Username,
+		Organizations: organizations,
+		Roles:         roles,
+		Audience:      []string{codeData.ClientID}, // Set audience to client ID from authorization request
+		Issuer:        s.config.Issuer,             // Set issuer
+	}
+	// Generate access token with proper expiry (1 hour)
+	accessToken, err := s.jwtGenerator.GenerateTokenWithType(tokenGenerationRequest, time.Hour, "access_token")
+	if err != nil {
+		return &pamapi.TokenResponse{Error: lo.ToPtr("server_error")}, nil
+	}
+
+	// Create token response
+	tokenResponse := &pamapi.TokenResponse{
+		AccessToken: lo.ToPtr(accessToken),
+		TokenType:   lo.ToPtr(pamapi.Bearer),
+		ExpiresIn:   lo.ToPtr(int(time.Hour.Seconds())),
+	}
+
+	// Only generate refresh token if offline_access was requested
+	if strings.Contains(codeData.Scope, "offline_access") {
+		refreshToken, err := s.jwtGenerator.GenerateTokenWithType(tokenGenerationRequest, 7*24*time.Hour, "refresh_token")
+		if err != nil {
+			return &pamapi.TokenResponse{Error: lo.ToPtr("server_error")}, nil
+		}
+		tokenResponse.RefreshToken = lo.ToPtr(refreshToken)
+	}
+
+	return tokenResponse, nil
+}
+
+// Authorize handles the authorization endpoint for authorization code flow
+func (s *PAMOIDCProvider) Authorize(ctx context.Context, req *pamapi.AuthAuthorizeParams) (*AuthorizeResponse, error) {
+	s.log.Infof("Authorize: starting authorization request - ClientId=%s, RedirectUri=%s, ResponseType=%s, Scope=%s, State=%s",
+		req.ClientId, req.RedirectUri, req.ResponseType, lo.FromPtrOr(req.Scope, ""), lo.FromPtrOr(req.State, ""))
+
+	// Validate required fields
+	if req.ClientId == "" || req.RedirectUri == "" {
+		s.log.Warnf("Authorize: missing required fields - ClientId=%s, RedirectUri=%s", req.ClientId, req.RedirectUri)
+		return nil, fmt.Errorf("invalid_request")
+	}
+
+	// Validate client ID
+	if s.config == nil {
+		s.log.Warnf("Authorize: config is nil")
+		return nil, fmt.Errorf("invalid_client")
+	}
+	if s.config.ClientID != req.ClientId {
+		s.log.Warnf("Authorize: invalid client ID - expected=%s, got=%s", s.config.ClientID, req.ClientId)
+		return nil, fmt.Errorf("invalid_client")
+	}
+	s.log.Infof("Authorize: client ID validation passed - %s", req.ClientId)
```

Comment:
same question here probably for all info logs

Reply:
fixed